obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+                caz
+    # Write event to catalog object for ease of .write() method
+    return new_event
+
+
+def readwavename(sfile):
+    """
+    Extract the waveform filename from the s-file.
+    Returns a list of waveform names found in the s-file as multiples can \
+    be present.
+
+    :type sfile: str
+    :param sfile: Path to the sfile
+
+    :returns: List of strings of wave paths
+    :rtype: list
+    """
+    f = open(sfile, 'r')
+    wavenames = _readwavename(f=f)
+    f.close()
+    return wavenames
+
+
+def _readwavename(f):
+    """
+    Internal wavename reader.
+
+    :type f: file
+    :param f: File open in read-mode
+    :return: list of wavefile names
+    """
+    wavename = []
+    for line in f:
+        if len(line) == 81 and line[79] == '6':
+            wavename.append(line[1:79].strip())
+    return wavename
+
+
+def blanksfile(wavefile, evtype, userid, outdir, overwrite=False,
+               evtime=False):
+    """
+    Generate an empty s-file with a populated header for a given waveform.
+
+    :type wavefile: str
+    :param wavefile: Wavefile to associate with this S-file, the timing of \
+        the S-file will be taken from this file if evtime is not set.
+    :type evtype: str
+    :param evtype: Event type letter code, e.g. L, R, D
+    :type userid: str
+    :param userid: 4-character SEISAN USER ID
+    :type outdir: str
+    :param outdir: Location to write S-file
+    :type overwrite: bool
+    :param overwrite: Overwrite an existing S-file, default=False
+    :type evtime: obspy.core.utcdatetime.UTCDateTime
+    :param evtime: If given this will set the timing of the S-file
+
+    :returns: str, S-file name
+    """
+
+    from obspy import read as obsread
+    import os
+    import datetime
+
+    if not evtime:
+        try:
+            st = obsread(wavefile)
+            evtime = st[0].stats.starttime
+        except:
+            raise IOError('Wavefile: ' + wavefile +
+                          ' is invalid, try again with real data.')
+    # Check that user ID is the correct length
+    if len(userid) != 4:
+        raise IOError('User ID must be 4 characters long')
+    # Check that outdir exists
+    if not os.path.isdir(outdir):
+        raise IOError('Out path does not exist, I will not create this: ' +
+                      outdir)
+    # Check that evtype is one of L,R,D
+    if evtype not in ['L', 'R', 'D']:
+        raise IOError('Event type must be either L, R or D')
+
+    # Generate s-file name in the format dd-hhmm-ss[L,R,D].Syyyymm
+    sfile = outdir + '/' + str(evtime.day).zfill(2) + '-' +\
+        str(evtime.hour).zfill(2) +\
+        str(evtime.minute).zfill(2) + '-' +\
+        str(evtime.second).zfill(2) + evtype + '.S' +\
+        str(evtime.year) +\
+        str(evtime.month).zfill(2)
+    # Check is sfile exists
+    if os.path.isfile(sfile) and not overwrite:
+        print('Desired sfile: ' + sfile + ' exists, will not overwrite')
```

Comment:
Sounds good :-) As long as the `Catalog.write` behaves in line with all other file formats and the overwriting behavior is documented I'm fine with it. Especially if this is something that people coming from Seisan will expect.